### DISCUSSION TITLE
How to get all the sub-sub-views in an array?

### DISCUSSION THREAD

**Discussion by frjnn** [rust-ndarray/ndarray#1003] [Q&A]
Given a four dimensional array of shape (X, Y, Z, W) is there a concise way to get all the (X x Y) 2-dimensional sub-sub-views of shape (Z,W)? (or all the the (A x B) 3-dimensional sub-sub-views of shape (C, D, E) in a 5-dimensional array of shape (A, B, C, D, E)

Addendum: currently I'm doing it by reshaping to (XY,  Z, W) and using `axis_iter`.

**Comment by bluss** [comment]
I don't know what the concisest way is, but there are methods for slicing (can also be used for indexing) and there's an indices producer that can let you iterate over the X x Y shape.
